src/rpc/claimtrie.cpp

```diff
@@ -180,118 +216,112 @@ bool getValueForClaim(const COutPoint& out, std::string& sValue)
+    for (std::vector<CSupportValue>::const_iterator itSupports = supports.begin(); itSupports != supports.end(); ++itSupports) {
+        supportObjs.push_back(supportToJSON(*itSupports));
     }
     ret.push_back(Pair("claimId", itClaimsAndSupports->first.GetHex()));
     ret.push_back(Pair("txid", claim.outPoint.hash.GetHex()));
     ret.push_back(Pair("n", (int)claim.outPoint.n));
     ret.push_back(Pair("nHeight", claim.nHeight));
     ret.push_back(Pair("nValidAtHeight", claim.nValidAtHeight));
-    if (claim.nValidAtHeight < nCurrentHeight)
-    {
-        nEffectiveAmount += claim.nAmount;
-    }
     ret.push_back(Pair("nAmount", claim.nAmount));
-    std::string sValue;
-    if (getValueForClaim(claim.outPoint, sValue))
-    {
-        ret.push_back(Pair("value", sValue));
-    }
     ret.push_back(Pair("nEffectiveAmount", nEffectiveAmount));
     ret.push_back(Pair("supports", supportObjs));
-
-    return ret;
-}
-
-UniValue supportsWithoutClaimsToJSON(supportsWithoutClaimsMapType::const_iterator itSupportsWithoutClaims, int nCurrentHeight)
-{
-    const std::vector<CSupportValue>& supports = itSupportsWithoutClaims->second;
-    UniValue ret(UniValue::VOBJ);
-    UniValue supportObjs(UniValue::VARR);
-    ret.push_back(Pair("claimId (no matching claim)", itSupportsWithoutClaims->first.GetHex()));
-    for (std::vector<CSupportValue>::const_iterator itSupports = supports.begin(); itSupports != supports.end(); ++itSupports)
-    {
-        UniValue supportObj(UniValue::VOBJ);
-        supportObj.push_back(Pair("txid", itSupports->outPoint.hash.GetHex()));
-        supportObj.push_back(Pair("n", (int)itSupports->outPoint.n));
-        supportObj.push_back(Pair("nHeight", itSupports->nHeight));
-        supportObj.push_back(Pair("nValidAtHeight", itSupports->nValidAtHeight));
-        supportObj.push_back(Pair("nAmount", itSupports->nAmount));
-        supportObjs.push_back(supportObj);
-    }
-    ret.push_back(Pair("supports", supportObjs));
     return ret;
 }
 
 UniValue getclaimsforname(const UniValue& params, bool fHelp)
 {
-    if (fHelp || params.size() != 1)
+    if (fHelp || params.size() > 2)
         throw std::runtime_error(
             "getclaimsforname\n"
             "Return all claims and supports for a name\n"
             "Arguments: \n"
-            "1.  \"name\"       (string) the name for which to get claims and supports\n"
+            "1.  \"name\"         (string) the name for which to get claims and supports\n"
+            "2.  \"blockhash\"    (string, optional) the hash of the block\n"
+            "                                           which is the active\n"
+            "                                           in the chain. If\n"
+            "                                           none is given, \n"
+            "                                           the latest block\n"
+            "                                           will be used.\n"
             "Result:\n"
```

Comment:
how about, "get claims for name at the block specified by this block hash. If none is given, the latest active block will be used"